src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,241 @@
+     */
+
+    public void handleTaskWithTime(String type, String description) {
+        String[] text = description.split(" ");
+        String task = text[0]; //initialise with first word of task
+        int indexOfTime = 0;
+        for (int i = 1; i < text.length; i++) { //start loop from second word of task
+            if (text[i].contains("/")) {
+                indexOfTime = i;
+                break;
+            }
+            task = task + " " + text[i];
+        }
+        indexOfTime++;
+        String time = "";
+        for (int i = indexOfTime; i < text.length; i++) {
+            if (i < text.length-1) { //if it is the last word
+                time = time + text[i] + " ";
+            } else {
+                time += text[i];
+            }
+        }
+        if (type.equals("deadline")) {
+            Task newTask = new Deadline(task, time);
+            tasks.add(newTask);
+        } else if (type.equals("event")) {
+            Task newTask = new Event(task, time);
+            tasks.add(newTask);
+        }
+    }
+
+    /**
+     * Processes user commands concerning tasks, irrespective of whether they have a time-related attribute.
+     * After processing, the task is created and added to the user's list of tasks.
+     *
+     * @param type The type of task being handled by the program.
+     * @param description The full description of the task being handled.
+     */
+    public void handleTask(String type, String description) {
+        if (type.equals("todo")) {
+            Task newTask = new Todo(description);
+            tasks.add(newTask);
+        } else {
+            this.handleTaskWithTime(type, description);
+        }
+    }
+
+    /**
+     * Processes the user input to the program and adds the processed task to the user's list of tasks if valid.
+     *
+     * @param type The type of task being handled by the program.
+     * @param curr The string representing the task currently being handled, including its type and description (if any).
+     * @throws EmptyDescriptionException if the user did not enter a description along with the type.
+     * @throws UnknownCommandException if the user provided an invalid type of task that the program cannot recognise.
+     */
+    public void addTask(String type, String curr) throws EmptyDescriptionException, UnknownCommandException {
+        if (!(type.equals("todo") || type.equals("deadline") || type.equals("event"))) { //type given is invalid
+            throw new UnknownCommandException();
+        } else {
+            String[] text = curr.split(" ");
+            if (text.length == 1) { //description is empty
+                throw new EmptyDescriptionException();
+            } else {
+                String description = assembleTaskDescription(text);
+                handleTask(type, description);
+                Ui.printAddTaskAcknowledgement(this.tasks, this.tasksCount);
+                this.tasksCount++;
+            }
```

Comment:
[Code Quality] I think the happy path could be make more prominent. Figure out how?